### PR TITLE
Add GitHub authorization token and requester

### PR DESCRIPTION
**Pull request by youngyojun**
This revision includes:
* Add `Token` class for GitHub user's personal access token
* Add `Requester` class which requests [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) queries with given authorization token
  * `Session` is an interface of a request session.
  * `RequestsSession` uses `requests.session` for actual uses, whereas `MockSession` uses its own class for unit tests and debugging.
* Unit tests for each class

### CODE REVIEW

File: src/polytope/main.py [hunk 1]
```diff
@@ -1,2 +1,16 @@
...
+    global GITHUB_TOKEN
+
+    load_dotenv()
+    GITHUB_TOKEN = os.environ.get("GITHUB_TOKEN")
+
+
 if __name__ == "__main__":
-    print("Hello, world!")
+    load_environment()
+    print(GITHUB_TOKEN)
```
Comment (TAMREF) [inline src/polytope/main.py]: is it intended?

File: src/polytope/github/Requester.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
+from typing import TYPE_CHECKING, Any, Optional, List, Callable, Protocol
+from enum import Enum, auto
+
+import requests
+
+if TYPE_CHECKING:
```
Comment (TAMREF) [inline src/polytope/github/Requester.py:9]: (comment) Please leave a comment about purpose of this clause. (only for mypy-testing)

File: src/polytope/github/Requester.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    def request(
+        self,
+        method: RequestMethod,
+        api_url: str,
+        **kwargs,
+    ) -> requests.Response:
+        """! API request wrapper with token authorization.
+
+        @param method   A HTTPS method.
+        @param api_url  A relative URL of API.
```
Comment (TAMREF) [inline src/polytope/github/Requester.py]: (comment) Do we need to enforce api_url must be start with `/`? If so, do we need a clarification in comments?

File: src/polytope/github/Requester.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    GET = auto()
+    HEAD = auto()
+    POST = auto()
+    PUT = auto()
+    DELETE = auto()
+    PATCH = auto()
+
+
+class Requester:
```
Comment (TAMREF) [inline src/polytope/github/Requester.py]: (suggestion) No need to do so, but how about defining an abstract class `Requestable` to separate `ActualRequester` and `MockRequester`? It's very new to me to inject methods into a class working itself.

**Review by TAMREF — COMMENTED** [review]
Thanks for the work! Looks good to me overall. Let's fix the test error.

File: src/polytope/github/Requester.py (hunk 2 above)
Comment (youngyojun) [reply src/polytope/github/Requester.py:9]: I left a comment about the `TYPE_CHECKING` clause.

https://github.com/Studio-Polytope/Polytope/blob/e47462fa5d8c0aa896e6965be70149c2f90994f1/src/polytope/github/Requester.py#L4-L8

File: src/polytope/github/Requester.py (hunk 3 above)
Comment (youngyojun) [reply src/polytope/github/Requester.py]: Yes, we do so.
I enforced that `api_url` must start with `/` and asserted it.

https://github.com/Studio-Polytope/Polytope/blob/e47462fa5d8c0aa896e6965be70149c2f90994f1/src/polytope/github/Requester.py#L59-L75

File: src/polytope/main.py (hunk 1 above)
Comment (youngyojun) [reply src/polytope/main.py]: Just hope to show how we treat an user's token.

File: src/polytope/github/Requester.py (hunk 4 above)
Comment (youngyojun) [reply src/polytope/github/Requester.py]: Looks like the best answer to me. :D

I will define an ABC `Requestable` and its subclasses `Requester` and `MockRequester`.

**Review by utilForever — APPROVED** [review]
I judged that the purpose of code `print` was for development. Please delete it later.

File: src/polytope/main.py (hunk 1 above)
Comment (youngyojun) [reply src/polytope/main.py]: As @utilForever said, the `print` code is removed.

https://github.com/Studio-Polytope/Polytope/blob/9a9d5fcd66a704086239b5784a6e93b2cebcf4ec/src/polytope/main.py#L13-L15

File: src/polytope/github/Requester.py (hunk 4 above)
Comment (youngyojun) [reply src/polytope/github/Requester.py]: I defined `Session` interface, and its child classes `RequestsSession` and `MockSession`.

`Requester` takes `SessionClass` as its initialization argument, and `MockSession` is used (only) for unit tests.

https://github.com/Studio-Polytope/Polytope/blob/7c3a45debd53cc98d28dca0098473e52d6b9c74e/src/polytope/github/Requester.py#L220-L242

**Review by TAMREF — APPROVED** [review]
LGTM. Thank you for addressing my comments.

**Pull request merged**